atomic_reactor/plugins/exit_koji_promote.py

```diff
@@ -326,43 +332,93 @@ def get_buildroot(self, build_id):
 
         return buildroot
 
+    def parse_orchestrator_logs(self, logs):
+        """
+        Parse the output of get_orchestrator_build_logs()
+
+        :return: list, Output instances
+        """
+        output = []
+
+        # Build list of arches
+        arches = []
+        for log in logs:
+            if log.platform is not None and log.platform not in arches:
+                arches.append(log.platform)
```

Comment:
In fact, we should avoid iterating more than once. It's a generator so further iterations will produce no items.

How about
```
platform_logs = {}
for log in logs:
    platform_logs.setdefault(log.platform, [])
    platform_logs[log.platform].append(log.line)
```
etc